apisix/plugins/log-rotate.lua

```diff
@@ -299,23 +313,34 @@ local function rotate()
     end
 
     if now_time >= rotate_time then
-        local files = {DEFAULT_ACCESS_LOG_FILENAME, DEFAULT_ERROR_LOG_FILENAME}
+        local new_access_log, new_error_log = get_custom_logfile_name()
+        
+        local access_log_filename = new_access_log or DEFAULT_ACCESS_LOG_FILENAME 
+        local error_log_filename = new_error_log or DEFAULT_ERROR_LOG_FILENAME
+        
+        local files = {access_log_filename, error_log_filename}
+
         rotate_file(files, now_time, max_kept)
 
         -- reset rotate time
         rotate_time = rotate_time + interval
 
     elseif max_size > 0 then
-        local access_log_file_size = file_size(default_logs[DEFAULT_ACCESS_LOG_FILENAME].file)
-        local error_log_file_size = file_size(default_logs[DEFAULT_ERROR_LOG_FILENAME].file)
+        local new_access_log, new_error_log = get_custom_logfile_name()
+        
+        local access_log_filename = new_access_log or DEFAULT_ACCESS_LOG_FILENAME 
+        local error_log_filename = new_error_log or DEFAULT_ERROR_LOG_FILENAME
+
+        local access_log_file_size = file_size(default_logs[access_log_filename].file)
+        local error_log_file_size = file_size(default_logs[error_log_filename].file)
         local files = core.table.new(2, 0)
 
         if access_log_file_size >= max_size then
-            core.table.insert(files, DEFAULT_ACCESS_LOG_FILENAME)
+            core.table.insert(files, access_log_filename)
         end
 
         if error_log_file_size >= max_size then
-            core.table.insert(files, DEFAULT_ERROR_LOG_FILENAME)
+            core.table.insert(files, error_log_filename)
```

Comment:
Does this seem fine? @monkeyDluffy6017

Reply:
The `access_log_filename ` and `error_log_filename ` could be defined out of the function, and be a module variable.
we could assign the value to them in `init` function

Reply:
Do you mean the `_M.init()` function?

Reply:
Yes

Reply:
Would creating these variables as module variables work since I can't define it at the top of the file as `get_custom_logfile_name` is defined in line 273

Reply:
First define the variable at the top of the file.
Then assign a value to this variable in the `init` function

Reply:
Done